### PR TITLE
Rework backend logic

### PR DESCRIPTION
**Pull request by EnQs**
Remove code mess from backend and make all it's features work properly.
Move all logic to backend.

### CODE REVIEW

File: src/EnQs/backend/Backend.java [hunk 1]
```diff
@@ -6,52 +6,125 @@
...
+    }
+
+    private void resetLock() {
+        lockedOperation = KeyProperties.PLUS.getOperation();
+        lockedValue = 0.0;
+    }
+
+    private static String getFormattedString(Double a) {
+        String format = String.format("%s", a);
+        format = format.substring(format.length() - 2).equals(".0") ? format.substring(0, format.length() - 2) : format;
```
Comment (Tyvrel) [inline src/EnQs/backend/Backend.java]: use `endsWith`

File: src/EnQs/backend/Edition.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package EnQs.backend;
+
+import java.util.function.Function;
+
+public enum Edition {
```
Comment (Tyvrel) [inline src/EnQs/backend/Edition.java]: Those should be Operation too, right?

File: src/EnQs/backend/Operation.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package EnQs.backend;
+
+import java.util.function.BiFunction;
+
+public enum Operation {
```
Comment (Tyvrel) [inline src/EnQs/backend/Operation.java]: enum ends by `Type`, so `OperationType`

File: src/EnQs/backend/Backend.java [hunk 4]
```diff
@@ -6,52 +6,125 @@
...
+        currentValue = lastCommand.getFunctionType() == FunctionType.EDITOR
+                || lastCommand.getFunctionType() == FunctionType.NUMERICAL
+                || lastCommand.getFunctionType() == FunctionType.MEMORY
+                ? currentValue + currentCommand.getOperation() : currentCommand.getOperation();
+    }
+
+    private void mathematicalKeyReact() {
+        Double numberEdited = Double.valueOf(currentValue);
+        lockedValue = operationHashMap.get(lockedOperation).apply(lockedValue, numberEdited);
+        lockedOperation = currentCommand.getOperation();
```
Comment (Tyvrel) [inline src/EnQs/backend/Backend.java]: Change type of `lockedOperation` to `OperationType`
`lockedOperation = OperationType.valueOf(currentCommand.getOperation());`

File: src/EnQs/backend/Backend.java (hunk 1 above)
Comment (EnQs) [reply src/EnQs/backend/Backend.java]: Ok, done.

File: src/EnQs/backend/Edition.java (hunk 2 above)
Comment (EnQs) [reply src/EnQs/backend/Edition.java]: Opperation and Eddition have deferent arguments

File: src/EnQs/backend/Operation.java (hunk 3 above)
Comment (EnQs) [reply src/EnQs/backend/Operation.java]: Done

File: src/EnQs/backend/Backend.java (hunk 4 above)
Comment (EnQs) [reply src/EnQs/backend/Backend.java]: Simplified fields

**Pull request merged**